src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadContestedCaseDocumentsService.java

```diff
@@ -0,0 +1,221 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_CORRESPONDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_EVIDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_FR_FORM_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_TRIAL_BUNDLE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_UPLOADED_DOCUMENTS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_CORRESPONDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_EVIDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_FR_FORM_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_TRIAL_BUNDLE_COLLECTION;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UploadContestedCaseDocumentsService {
+
+    private static final String APPLICANT = "applicant";
+    private static final String RESPONDENT = "respondent";
+
+    private final ObjectMapper mapper;
+
+    public Map<String, Object> filterDocumentsToRelevantParty(Map<String, Object> caseData) throws Exception {
+
+        List<ContestedUploadedDocumentData> uploadedDocuments = getDocumentCollection(caseData, CONTESTED_UPLOADED_DOCUMENTS);
+
+        filterApplicantCorrespondence(uploadedDocuments, caseData);
+        filterApplicantForms(uploadedDocuments, caseData);
+        filterApplicantEvidence(uploadedDocuments, caseData);
+        filterApplicantTrialBundle(uploadedDocuments, caseData);
+
+        filterRespondentCorrespondence(uploadedDocuments, caseData);
+        filterRespondentForms(uploadedDocuments, caseData);
+        filterRespondentEvidence(uploadedDocuments, caseData);
+        filterRespondentTrialBundle(uploadedDocuments, caseData);
+
+        caseData.put(CONTESTED_UPLOADED_DOCUMENTS, uploadedDocuments);
+
+        return caseData;
+    }
+
+    private List<ContestedUploadedDocumentData> getDocumentCollection(Map<String, Object> caseData, String collection) {
+
+        if (StringUtils.isEmpty(caseData.get(collection))) {
+            return new ArrayList<>();
+        }
+
+        return mapper.convertValue(caseData.get(collection),
+            new TypeReference<List<ContestedUploadedDocumentData>>() {
+            });
+    }
+
+    private boolean isTypeValidForCorrespondence(String caseDocumentType) {
+        return caseDocumentType.equals("Letter from Applicant");
+    }
+
+    private boolean isTypeValidForForms(String caseDocumentType) {
```

Comment:
do we need to validate the types? if these are all that they can select in CCD then we shouldn't need to validate. Also if we ever play a story to add a new item to the list in CCD we'll also need to come back here and update although we'll probably forget about this and underestimate

Reply:
We need to "validate" the forms in order to determine which document header they fall into, for example:

	FR Forms
		-> Form B
		-> Form E ...
	Evidence In Support:
		-> Statement of Issues
		-> Chronology...